src/streetscapes/sources/image/mapillary.py

```diff
@@ -77,103 +166,224 @@ def create_session(self) -> requests.Session:
             A `requests` session.
         """
 
-        session = requests.Session()
+        session = Session()
         session.headers.update({"Authorization": f"OAuth {self.token}"})
+        retries = Retry(
+            total=3,
+            backoff_factor=0.1,
+            status_forcelist=[500, 502, 503, 504],
+        )
+        session.mount("https://", HTTPAdapter(max_retries=retries))
         return session
 
-    def fetch_image_ids(
+    def collect_data(self, url, params, filename):
+        """Collect metadata records from the API and dump into a json file
+
+        Args:
+            url: url for downloading metadata
+            params: dictionary of parameters to extract
+            filename: filename for the json file
+
+        Returns:
+            json of metadata records from API
+        """
+        response = self.session.get(url, params=params)
+        response.raise_for_status()
+        data = response.json()
+        # Collect data
+        records = data.get("data", [])
+        with open(filename, "w") as f:
+            json.dump(records, f)
+        return data
+
+    def fetch_metadata_bbox(
         self,
         bbox: list[float],
+        tile_size: float = 0.01,
         fields: list[str] | None = None,
-        limit: int = 500,
-        extract_latlon: bool = True,
+        limit: int = 1000,
+        bbox_name: str = "bbox",
```

Comment:
Maybe we don't need the bbox_name anymore. Only coordinates of the tile is also fine, right?